# Introduction
Memote reports quite an extensive list of unbalances reactions. Martyn and I have split the list and will look into them by hand to resolve these issues as much as possible.

Here I have 'attached' the file of the total list, and included the split for me and martyn. 

In [1]:
import cameo
import pandas as pd
import cobra.io
import escher

In [2]:
model = cobra.io.read_sbml_model('../model/g-thermo.xml')

In [7]:
unbalanced_rct =["23DK5MPPISO","T_cbl1_e","LARGNHYDRO","ADCL","OCBT","PUNP2","PGK","T_trnaglu_e","PYRACTT","MTRI","R_r0082","PBUTT","T_gdpmann_e","DTPCUNPPT","OMCDC","ABUTD","DHQS","T_udpgalur_e","INO1PPHYDRO","T_adgppuc_e","MDRPD","OXGDC","T_pydx5p_e","KARI_23dhmb","NTD3","T_man_e","T_qh2_3","HBCOAOR","IDPh","ACENLAT","T_trp__L_e","HISOR","DHAD1","T_cyan_e","MHTPPT","T_ile__L_e","T_cmp_e","T_adprib_e","SUCRGT","ACM6PH","OCTACPOR","MOD_3mop","ACGK","T_dhgly_e","ALDD2x","AIZCLY","R_UDPGD","T_pyridoxal_e","T_uri_e","T_phe__L_e","T_ump_e","OCTACPAT","MHOBT","R_SSALx","T_chlac_e","2OH3K5MPPISO","T_abmaagapc_e","EN6LYOR","R_SSALy","ALASm","T_rib__D_e","T_din_e","ENCOAT","NNAM","3OAR100","T_mercplac_e","AMBALY","T_octdecacp_e","T_damp_e","MMM2","ACDO","QUINOR","OPALMACPOR","T_hepdp_e","HDEACPT","DATPT","PUNP1","OAADC","T_gmagg_e","3OAR40_1","ADOCBIP","PTAr","HMBS","XU5PGT","T_but_e","T_gal_e","MALS","ACS","4ABUTD","ATPPT","DHFS","DURIPP","T_co2_e","R_SERD_L","DGTPT","S7PTRANSPP","3HAD40_1","DHORTS","T_betaine_e","URAHL","T_amet_e","T_drib_e","GLNAMHYDRO","T_ps_cho_e","PENTOHYDROARA","PCPPTh","HMSD","R_SELCYSTL","UAAAGGLAAP","T_val__L_e","T_M_10fthf_e","IVCOAR","DHBS","ASAD","T_slnt_e","T_catechol_e","QH2NOR","R_r0068","GCCa","ACMUMPT","PDHam1hi","PHSERT","F1PPT","L23DHDPICOLY","MME","GMPS2","T_accoa_e","ILEOR","PSP","T_phe__D_e","T_his__L_e","ATPNAT","T_etoh_e","GLCDPT","TREPT","ACHMSSELCYSL","NTD5","T_pheme_e","PHEOR","T_gln__D_e","G6PI_1","R_SUCHMSSELCYSL","FTHFLIG","MMSAD1","NTD6","BCFASYN3","IPPS","CHLACOR","AGGPR","T_met__L_e","NTD11","R_UPP1S","PRPAT","MALTPT","CCP","T_a4oxopent_e","T_adn_e","APLh","AMYTRANS","ACBIPGT","GMP5N","G5DHx","STARCHGT","T_fe2_e","T_lac__L_e","T_M_C04051_e","ACAS_2ahbut","FICYTCCNOR","T_prpncoa_e","HC01435R","NO2R","T_amp_e","T_bcfa_e","R_UACMAMO","ADOCBLS","AMETDMMQ","T_oh1_e","ABFPT","COOR","T_csn_e","HMCNSADLY","ARGDC","HGBYR","PUNP3","MANPT","T_uamagdgll_e","MENAOR","G6PI2","PPAPPT","T_M_3hmbcoa_e","T_gmp_e","R_TDP","MAN1PT","QH2OR","T_thmtp_e","NT5C","GAMPT","PRPNTEAT","ACDAH","T_dmbzid_e","T_o2_e","PC20M","R_TALA","UAAAGGLAAT","T_mesel_e","PUNP6","T_fum_e","GLU5K","GDH_nadp","T_chor_e","T_nadph_e","R_TKT2","CELLOHYDRO","T_sprm_e","DECOT","XYLANHYDRO","R_UAGCVT","DECACPOR","T_nadp_e","R_SHSL4h","SELMELY","DECACPAT","T_arg__L_e","GAM","DECDPPP","GDH","T_M_2hymeph_e","ADCPS2","2HBO","T_malcoa_e","T_mmal_e","PUTA3","T_hpglu_e","R_SBP","AUS","DHNAOCTD","ACGS","T_decanoylacp_e","T_fol_e","T_dhptd_e","R_THRPD","SELMELIG","FBAf","UAGUDPGPTRANS","AAGAMPT","R_TMDPPK","DALAOX","T_cpppg1_e","GAPD","R05219","MERCPPYRST","COCHL","T_appl_e","NTD8","GAPDH_nadp_hi","T_thrp_e","ACKr","PUNP4","SCUGLUTRA","XYLHYDRO","PC6YM","HEX7","RAFFINHYDRO","R_VALDHr","R_r0097","GTPCI","LARGNOR","T_cobalt2_e","G5SADrm","R_UPPDC1","NNDPR","GLYALDDr","T_nh4_e","NTD10","MOD","T_lac__D_e","KAS14","FDPPH","AHSL","MMTSAOR","MHOPT","G6PDA","LLEUDr","GDTP","T_for_e","MOD_4mop","T_glc__D_e","T_succ_e","MMALLIG","H202OR","T_acmum_e","ALAALALIG","NP1","R_r0775","PFKh","HACD6","DLDLBT","PYNP1","FPRAMLIG","BFBP","AKP1","CYANST","KARI_23dhmp","SAM","R_r0708","T_ser__L_e","R_SELCYSTGL","FGAMALIG","PUNP7","ICDHhr","PENTOHYDROXYL","KARI","T_selmethtrna_e","PRPUT","ALAALAr","THPPGLIG","IG3PS","T_dgmp_e","PRPGT","CTPS1","AMETTRANS","T_dlglu_c","QULNS","T_mthpglu_e","ACLS","ACCOAC","ASNN","ACSERL","PRPCT","FPGFTh","STACHY","METGL","HOPNTAL","T_M_5fthf_e","T_tcynt_e","CYSAT","ADSL2r","T_dcmp_e","ASP1DC","CTPS2","CHORS","NODOx","AHCYSMETTRANS","T_ftethplg_e","GTPT","CBPS","TRE6PHYDRO","T_cbi_e","T_hqn_e","NODOy","CTPT","FGFTh","T_dtmp_e","MALACPAT","SBZCOAHYDRO","ACYSAT","STACHY2","CYSTGL","T_glyc_e","R_r0163","SALCNPT","MMACPAT","GLYOR","CYSTT","T_tre_e","NAMNPP","FOMETRi","T_ura_e","GLUKA","R_SHSL1","HDH","HCYSMT","T_ala__L_e","STARCHHY","ALKP","PPCDC","FTCL","T_dodecacp_e","CYSDS","PPND","CDPDAGPT","T_dad_2_e","T_M_3omrsACP_e","PYRPT","T_asp__L_e","AHSERL","NMNHYD","T_but2eACP_e","GLYO1","GLYOX","NANEURATRANS","FOCYTCCOR","ADSS","ACPh","DHAD2","ACHBS","T_thf_e","PPCOAOm","HEXACPOR","CYSTHI","KARI_3hmoa","COQOR","HEXACPAT","FDH","R_UGE","BCFASYN","T_dtpcudcpp_e","GND","PPBNGS","ARBTPT","MECDPOR","CBL","PRPNTELIG","PPCOAC","T_glc6P_e","M1PD","2AMACHYD","AHHST","T_gam_e","MNLPT","IPDPCT","R_SHSL2","PAMPT","T_nad_e","PUNP5","PSCIT","DTPCUDCPPR","MAN6PI","T_hxan_e","T_M_5mthf_e","ADPGLCGT","PDH","NNDMBRT","GTPCII2","ASAPT","AMYLOSEGT","FBA3","MGSA","ANS","T_nadh_e","T_dcyt_e","DDPA","PYDXS","R_r0120","ME1","T_gthrd_e","AMETLY","R_TAL","ECOAH4","ASPO1","T_thymd_e","BGLUCH","T_aacoa_e","T_coa_e","PPCK","DCTPT","MSAOR","R_TMPPP","NTD7","APAPTm","T_inost_e","CYSS","UTPT","MALTHYDRO","T_capra_e","ATPM","DXPS","T_mlthf_e","SUCHY","DAPDC","PC","T_RNA_e","T_ac_e","T_gthox_e","SUCHY2","DTTPT","MMSAD3","NTD1","T_succcoa_c","T_hmgcoa_e","FRU1PT","T_hemeO_e","STARCHHY2","T_g3pc_e","CS","NADS1","CELLBHY","T_acetol_e","ALAD_L","SMBCOAOR","PC17M","T_dna_e","MYRACPOR","T_hdeacp_e","AACOAT","GLUTRR","R_SEPHCHCS","PGCM","DTPCUNPTRANS","T_acon_e","T_uacmamu_e","GLUTRS","OMPDC","MBTCOAT","T_thm_e","T_asn__L_e","T_pmtcoa_e","DTPCUNPT","HBCOAH","T_14dhncoa_e","GCCb","FOCYTCNOR","NTP3","R_UAMAS","HEPDPP","T_ribflv_e","T_arbt_e","NTD2","ASPCT","T_mmcoa__S_e","T_gly_e","BCFASYN2","R_UPP3MT","PPGPPDP","SELCYSLY","AKGDH","GALISO","T_menqui_e","T_salcn_e","T_ins_e","PPNDH","PPRGL","ACSERT","NTD4","R_UAMAGS","DB4PS","R_THRS","ACYP_2","HISTP","IGPS","T_h_e","T_glcn__D_e","ASPK","ASPSALY","ADCPS1","ARABINANHY","UAMAGDGLLLIG","R_SUCBZL","BUTACPAT","AH6PISO","SCENLAT","T_so3_e","ASPO6","PC11M","ADOCBLTRANS","PRAISh","ADOCBIK","MYOIPPH","UAMAGLIG","RZ5PP","FTHFD","FTETHLGPT","T_cl_e","ECOAH9ir","AIRC3","ASNS2","PRE3AOR","GHMT","PANOPTRANS","T_nac_e"]
len(unbalanced_rct)

570

In [8]:
martyn_reactions = unbalanced_rct[0:285]
len(martyn_reactions)

285

In [9]:
vivienne_reactions = unbalanced_rct[286:569]

In [18]:
#make a list of all of Viviennes reactions that exlclude transport
metabolic_rct_viv = []
for rct in model.reactions:
    if rct.id in vivienne_reactions:
        if rct.id[-1:] in 't':
            continue
        else: 
            metabolic_rct_viv.append(rct)
    else: continue

In [19]:
len(metabolic_rct_viv)

190

So there are 190 metabolic reactions that I need to check.

I wil convert this into a list in .csv format so that I can keep track of which reactions I've fixed and which not, and make comments about some of the ones I doubt about.

In this section, I will modify 5 reactions at a time, and then save & commit, to prevent too many commits cluttering the system.

In [33]:
#make lists to export
rct_id = []
rct_kegg = [] 
for rct in metabolic_rct_viv:
    rct_id.append(rct.id)
    rct_kegg.append(rct.name)
    

In [34]:
metabolic_rct_viv_df = pd.DataFrame({'ID':rct_id, 'KEGG': rct_kegg, 'Done': ' ' , 'Comments': ' '})
metabolic_rct_viv_df[0:5]

ID    KEGG Done Comments
0     PPBNGS  R00036              
1  FOCYTCCOR  R00082              
2       ATPM  R00086              
3       NTD7  R00183              
4      NADS1  R00189

In [36]:
metabolic_rct_viv_df.to_csv('../databases/Metabolic_rct_viv.csv')

In [66]:
rct = model.reactions.ME1

rct

<Reaction ME1 at 0x21cb24288c8>

In [67]:
for met in rct.metabolites:
    print (met.name, ':', met.formula)

CO2_CO2 : CO2
NADH_C21H29N7O14P2 : C21H29N7O14P2
Pyruvate_C3H4O3 : C3H4O3
H+_H : H
(S)-Malate_C4H4O5 : C4H4O5
NAD+_C21H28N7O14P2 : C21H28N7O14P2


In [ ]:
#fix formula 
model.metabolites..formula = ''
model.metabolites..name = ''

In [ ]:
#fix formula 
model.metabolites..formula = ''
model.metabolites..name = ''

In [ ]:
#fix formula 
model.metabolites..formula = ''
model.metabolites..name = ''

In [ ]:
#fix formula 
model.metabolites..formula = ''
model.metabolites..name = ''

In [ ]:
#fix formula 
model.metabolites..formula = ''
model.metabolites..name = ''

In [65]:
#fix formula nadh_c
model.metabolites.nadh_c.formula = 'C21H29N7O14P2'
model.metabolites.nadh_c.name = 'NADH_C21H29N7O14P2'

In [64]:
#fix formula accoa_c
model.metabolites.accoa_c.formula = 'C23H38N7O17P3S'
model.metabolites.accoa_c.name = 'Acetyl-CoA_C23H38N7O17P3S'

In [63]:
#fix formula coa_c
model.metabolites.coa_c.formula = 'C21H36N7O16P3S'
model.metabolites.coa_c.name = 'CoA_C21H36N7O16P3S '

In [62]:
#fix formula pyr_c
model.metabolites.pyr_c.formula = 'C3H4O3'
model.metabolites.pyr_c.name = 'Pyruvate_C3H4O3'

In [57]:
#fix formula 
model.metabolites.nad_c.formula = 'C21H28N7O14P2'
model.metabolites.nad_c.name = 'NAD+_C21H28N7O14P2'

In [58]:
#fix formula 
model.metabolites.ppi_c.formula = 'H4P2O7'
model.metabolites.ppi_c.name = 'Pyrophosphate_H4O7P2'

In [59]:
#fix formula dnad_c
model.metabolites.dnad_c.formula = 'C21H27N6O15P2'
model.metabolites.dnad_c.name = 'Deamino-NAD+_C21H27N6O15P2'

In [54]:
#fix formula amp_c
model.metabolites.amp_c.formula = 'C10H14N5O7P'
model.metabolites.amp_c.name = 'AMP_C10H14N5O7P'

In [42]:
#fix formula ppbng_c
model.metabolites.ppbng_c.formula = 'C10H14N2O4'
model.metabolites.ppbng_c.name = 'Porphobilinogen_C10H14N2O4'

In [49]:
#fix formula atp_c
model.metabolites.atp_c.formula = 'C10H16N5O13P3'
model.metabolites.atp_c.name = 'ATP_C10H16N5O13P3'

In [48]:
#fix formula pi_c
model.metabolites.pi_c.formula = 'H3PO4'
model.metabolites.pi_c.name = 'Orthophosphate_H3PO4'

In [50]:
#fix formula adp_c
model.metabolites.adp_c.formula = 'C10H15N5O10P2'
model.metabolites.adp_c.name = 'ADP_C10H15N5O10P2'

In [20]:
metabolic_rct_viv

[<Reaction PPBNGS at 0x21cb23e8f88>,
 <Reaction FOCYTCCOR at 0x21cb23f2508>,
 <Reaction ATPM at 0x21cb23f24c8>,
 <Reaction NTD7 at 0x21cb2417888>,
 <Reaction NADS1 at 0x21cb2415808>,
 <Reaction PDH at 0x21cb2424688>,
 <Reaction ME1 at 0x21cb24288c8>,
 <Reaction ACLS at 0x21cb242b7c8>,
 <Reaction ICDHhr at 0x21cb2456a48>,
 <Reaction H202OR at 0x21cb245c7c8>,
 <Reaction ACYP_2 at 0x21cb24716c8>,
 <Reaction NTP3 at 0x21cb2476108>,
 <Reaction PPGPPDP at 0x21cb247e348>,
 <Reaction PPCK at 0x21cb24801c8>,
 <Reaction PC at 0x21cb2483a08>,
 <Reaction CS at 0x21cb2485708>,
 <Reaction ASPO1 at 0x21cb248c4c8>,
 <Reaction GLYOR at 0x21cb2479808>,
 <Reaction GLYO1 at 0x21cb24942c8>,
 <Reaction DCTPT at 0x21cb249bb88>,
 <Reaction DTTPT at 0x21cb249b2c8>,
 <Reaction ALAD_L at 0x21cb249ba08>,
 <Reaction GTPCII2 at 0x21cb24b6c48>,
 <Reaction PRPUT at 0x21cb24c8288>,
 <Reaction PRPGT at 0x21cb24c6808>,
 <Reaction PRPCT at 0x21cb24c6fc8>,
 <Reaction GTPT at 0x21cb24c8c08>,
 <Reaction CTPT at 0x21cb24cb5c

In [22]:
model.reactions.GTPT.notes

{'SUBSYSTEM': 'X',
 'GENE_ASSOCIATION': '( RTMO03624 or RTMO05254 or RTMO04080 or RTMO04824 or RTMO04082 )',
 'KEGG ID': 'R00441',
 'ENZYME': '2.7.7.6 2.7.7.48',
 'NAME': 'GTP:RNA guanylyltransferase (RNA-directed); GTP:RNA guanylyltransferase (DNA-directed);',
 'DEFINITION': 'GTP + RNA ⇌ Diphosphate + RNA'}

In [ ]:
model.metabolites